packages/fx-core/src/plugins/resource/officeaddin/questions.ts

```diff
@@ -1,58 +1,76 @@
-import { MultiSelectQuestion, OptionItem, SingleSelectQuestion } from "@microsoft/teamsfx-api";
+import {
+  Inputs,
+  OptionItem,
+  SingleSelectQuestion,
+  TextInputQuestion,
+} from "@microsoft/teamsfx-api";
+import projectsJsonData from "./config/projectsJsonData";
+
+const jsonData = new projectsJsonData();
 
 export enum QuestionName {
-  ExampleSingleSelectQuestion = "example-single-select",
-  ExampleMultiSelectQuestion = "example-multi-select",
+  AddinLanguageQuestion = "addin-language",
+  AddinNameQuestion = "addin-name",
+  AddinTemplateSelectQuestion = "addin-template-select",
+  OfficeHostQuestion = "addin-host",
 }
 
-export const SingleSelectOptionOne: OptionItem = {
-  id: "option1",
-  label: "Option 1 label",
-  detail: "Option 1 detail",
-  groupName: "group1",
-};
-
-export const SingleSelectOptionTwo: OptionItem = {
-  id: "option2",
-  label: "Option 2 label",
-  detail: "Option 2 detail",
-  groupName: "group1",
-};
-
-export const SingleSelectOptionThree: OptionItem = {
-  id: "option3",
-  label: "Option 3 label",
-  detail: "Option 3 detail",
-  groupName: "group2",
-};
-
 // TODO: localize the strings
-export const ExampleSingleSelectQuestion: SingleSelectQuestion = {
+export const AddinTemplateSelectQuestion: SingleSelectQuestion = {
   type: "singleSelect",
-  name: QuestionName.ExampleSingleSelectQuestion,
-  title: "This is a single select question",
-  staticOptions: [SingleSelectOptionOne, SingleSelectOptionTwo, SingleSelectOptionThree],
-  default: SingleSelectOptionOne.id,
+  name: QuestionName.AddinTemplateSelectQuestion,
+  title: "Add-in template type",
+  staticOptions: jsonData
+    .getProjectTemplateNames()
+    .map((template) => ({ label: jsonData.getProjectDisplayName(template), id: template })),
+  default: "teams-manifest",
   placeholder: "This is placeholder",
 };
 
-export const MultiSelectOptionOne: OptionItem = {
-  id: "multi-option1",
-  label: "Option 1 label",
-  detail: "Option 1 detail",
+export const AddinNameQuestion: TextInputQuestion = {
+  type: "text",
+  name: QuestionName.AddinNameQuestion,
+  title: "Add-in name",
+  default: "office-addin",
 };
 
-export const MultiSelectOptionTwo: OptionItem = {
-  id: "multi-option2",
-  label: "Option 2 label",
-  detail: "Option 2 detail",
+export const AddinLanguageQuestion: SingleSelectQuestion = {
```

Comment:
We already have a question that asks for choosing programming language between js and ts.  Can we reuse it?

Reply:
The question I made uses dynamic options based on a json file that describes the different template options we have and what supported on each of them (including which languages are available).  Is that something that can be done with the existing question?

Reply:
If the json file is committed to the repo, I think the questions are actually statically determined at compile time. You can read the json file and generate the questions.

Reply:
I think I get what you mean . . . but I'm not sure about the implementation.  This question creates the options available (using dynamicOptions) based on the content of the json file (yes statically added to the repo).  How would I instead conditionally create the question (using your existing question)?  I'm guessing I would use a function that returns the question instead of this type in index.ts . . . but at that point I don't know if I want to ask the question or what the options are because it depends on the answer to an earlier question.